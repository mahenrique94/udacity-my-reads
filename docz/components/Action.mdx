---
menu: Components
name: Action
order: 1
route: /
---

import { Playground, PropsTable } from "docz"

import Action from "../components/Action"

# Action

<Playground>
    <Action>My Action</Action>
</Playground>

<PropsTable of={ Action }/>
