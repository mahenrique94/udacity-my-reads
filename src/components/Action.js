import PropTypes from "prop-types"
import React from "react"

import i18n from "i18n"

import { Card } from "react-bulma-components"

import Confirm from "./Confirm"
import Show from "./Show"

const Action = ({ action, show, text }) => (

    <Show only={ show }>
        <Confirm
            message={ i18n.t("messages.books.updateShelf") }
            onConfirm={ action }
            title={ i18n.t("titles.books.action") }
        >
            <Card.Footer.Item onClick={ action } renderAs="a">
                { text }
            </Card.Footer.Item>
        </Confirm>
    </Show>

)

Action.defaultProps = {
    show: false
}

Action.propTypes = {
    action: PropTypes.func.isRequired,
    show: PropTypes.bool,
    text: PropTypes.string.isRequired
}

export default Action
