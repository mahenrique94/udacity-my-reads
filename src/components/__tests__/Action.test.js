import React from "react"
import renderer from "react-test-renderer"

import Action from "../Action.js"

describe("Tests for Action component", () => {

    const requiredProps = {
        action: () => console.log("Testing"),
        text: "Testing"
    }

    it("Should rendering without crashing", () => {
        const tree = renderer.create(
            <Action { ...requiredProps }/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it("Should rendering with message when it was provided", () => {
        const tree = renderer.create(
            <Action { ...requiredProps } message="I'm a message"/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

})
