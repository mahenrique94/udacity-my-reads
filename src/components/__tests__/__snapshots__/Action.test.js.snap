// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Action component Should rendering with message when it was provided 1`] = `null`;

exports[`Tests for Action component Should rendering without crashing 1`] = `null`;
